content/blog/hacking-build-process.md

```diff
@@ -0,0 +1,260 @@
+kn-workflow gen-manifest --profile gitops --image $TARGET_IMAGE
+```
+
+After running this command, you should see these files in the manifests directory:
+
+```bash
+manifests
+├── 01-configmap_lock-flow-props.yaml
+└── 02-sonataflow_lock-flow.yaml
+```
+
+These files alone won't be sufficient for a successful deployment. We'll make additional changes after building the workflow image.
+
+For now, let's verify the `02-sonataflow_lock-flow.yaml` contains a reference to our workflow image (required for the gitops profile):
+
+```yaml
+  podTemplate:
+    container:
+      image: quay.io/orchestrator/demo-poc-kafka-logic:latest
+      resources: {}
+```
+
+## Building the Workflow Image Using Dockerfile
+
+Let's build the workflow image using podman on a Linux machine. Navigate to the `poc-kafka-logic-operator/callback-flow` directory and run:
+
+```bash
+podman build \
+  -f ../../orchestrator-demo/docker/osl.Dockerfile \
+  --tag $TARGET_IMAGE \
+  --platform linux/amd64 \
+  --ulimit nofile=4096:4096 \
+  --build-arg QUARKUS_EXTENSIONS="\
+    org.kie:kie-addons-quarkus-persistence-jdbc:9.102.0.redhat-00005,\
+    io.quarkus:quarkus-jdbc-postgresql:3.8.6.redhat-00004,\
+    io.quarkus:quarkus-agroal:3.8.6.redhat-00004, \
+    io.quarkus:quarkus-smallrye-reactive-messaging-kafka" \
+  --build-arg MAVEN_ARGS_APPEND="\
+    -DmaxYamlCodePoints=35000000 \
+    -Dkogito.persistence.type=jdbc \
+    -Dquarkus.datasource.db-kind=postgresql \
+    -Dkogito.persistence.proto.marshaller=false" \
+    .
+```
+
+Note that for this particular workflow, we needed to include an additional Quarkus extension: `io.quarkus:quarkus-smallrye-reactive-messaging-kafka`.
+
+Once built, push the image to your registry:
+
+```bash
+podman push $TARGET_IMAGE
+```
+
+## Editing the Manifests
+
+Now we need to enhance our manifests to make them production-ready.
+
+### Enable Persistence
+
+In production environments, we want to enable persistence for our workflow. This can be configured either at the platform level or per workflow. See the [Sonataflow documentation for enabling persistence](https://sonataflow.org/serverlessworkflow/latest/cloud/operator/using-persistence.html#configuring-persistence-using-the-sonataflowplatform-cr) for more details.
+
+For our workflow, persistence configuration is required since we included Quarkus persistence extensions in the build command.
+
+Edit `src/main/resources/manifests/02-sonataflow_lock-flow.yaml` and add this section at the same level as the `podTemplate`:
+
+```yaml
+  persistence:
+    postgresql:
+      secretRef:
+        name: sonataflow-psql-postgresql
+        userKey: postgres-username
+        passwordKey: postgres-password
+      serviceRef:
+        name: sonataflow-psql-postgresql
+        port: 5432
+        databaseName: sonataflow
+        databaseSchema: lock-flow
+```
+
+Also, add the following property to the configmap to enable Flyway migrations:
```

Comment:
Suggestion: put the name of the configmap file `01-configmap_lock-flow-props.yaml` :
```
Also, add the following property to the configmap manifest `01-configmap_lock-flow-props.yaml` to enable Flyway migrations
```